src/NuGetForUnity/Editor/NugetPackageInstaller.cs

```diff
@@ -170,9 +171,36 @@ private static bool Install(INugetPackage package, bool refreshAssets, bool inst
                     // unzip the package
                     using (var zip = ZipFile.OpenRead(cachedPackagePath))
                     {
+                        var libs = new Dictionary<string, List<ZipArchiveEntry>>();
+
                         foreach (var entry in zip.Entries)
                         {
-                            var filePath = Path.Combine(baseDirectory, entry.FullName);
+                            var entryFullName = entry.FullName;
+                            var filePath = Path.Combine(baseDirectory, entryFullName);
+
+                            if (ShouldSkipUnpackingOnPath(entryFullName, package.Id))
+                            {
+                                continue;
+                            }
+
+                            // we don't want to unpack all lib folders and then delete all but one; rather we will decide the best
+                            // target framework before unpacking, but first we need to collect all lib entries from zip
+                            if (entryFullName.StartsWith("lib"))
+                            {
+                                const int frameworkStartIndex = 4; // length of "lib/"
+                                var secondSlashIndex = entryFullName.IndexOf("/", frameworkStartIndex, StringComparison.Ordinal);
+                                var framework = entryFullName.Substring(frameworkStartIndex, secondSlashIndex - frameworkStartIndex);
+                                if (libs.ContainsKey(framework))
```

Comment:
Maybe this can be written a bit better:
```c#
if (libs.TryGet(framework, out var entryList)
{
    entryList.Add(entry);
}
else
{
    libs[framework] = new List<ZipArchiveEntry> { entry };
}
```